### ISSUE TITLE
Bug in tools/qeinputparser/str2val

### ISSUE THREAD

**Issue by chrisjsewell** [aiidateam/aiida-quantumespresso#197]
The string '1e-08' is not accepted, i.e. the regex needs to accept an integer before the exponent. I believe this should do the trick:

```python
float_re = re.compile(r"""
        [-+]?               # optional sign
        (?:                   # either
        \d*[.]\d+           # 10.53 or .53
        |                       # or
        \d+[.]\d*           # 10.53 or 10.
        |                       # or
        \d+ )                 # an integer
    (?:[dEeE][-+]?[0-9]+)?  # optional exponent
    """, re.X)

```

**Comment by chrisjsewell** [comment]
I imagine this doesn't normally crop up in the output files, but its present in my input file when using `PwimmigrantCalculation`

**Comment by chrisjsewell** [comment]
fixed in #329
